data/extension/init/templates/tinygo/envoy.access_loggers/default/main_test.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxytest"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/types"
+)
+
+func TestAccessLogger_OnLog(t *testing.T) {
+	opt := proxytest.NewEmulatorOption().
+		WithNewRootContext(newAccessLogger)
+	host := proxytest.NewHostEmulator(opt)
+	// Release the host emulation lock so that other test cases can insert their own host emulation.
+	defer host.Done()
+
+	// Call OnLog with the given headers.
+	host.CallOnLogForAccessLogger(types.Headers{
```

Comment:
yeah. there *was* a reason for this: For http and network filter, I emulated the exact order of execution of OnLog, OnDone and OnDelete (though OnDelete is only and internally used in the SDK) in the Envoy: 
- nework: https://github.com/tetratelabs/proxy-wasm-go-sdk/blob/df9a0b0b2c9e5ce1733cffa34f4ea85b63be4ac9/proxytest/network.go#L133-L139
- http: https://github.com/tetratelabs/proxy-wasm-go-sdk/blob/df9a0b0b2c9e5ce1733cffa34f4ea85b63be4ac9/proxytest/http.go#L442-L450

where OnLog is called on each context corresponding to each stream underneath. That means I intentionally do not allow users to execute OnLog alone. 

For access logger, OnDone is called on RootContext and OnDone is not called so that's why I had this interface.


But now I feel that having both OnDone and OnLog in the http and network context is a bit confusing and exposing only OnLog would be much better because OnDone is called soon after/before in network/http filter and always return true). Will fix the interface.

Reply:
Well, I couldn't find the better way, let me come back to this later and leave as is for now, since the next ABI (Piotr said it will be shared by the end of this month) will drastically change and it is likely that I have to refactor test framework anyway.